luarules/gadgets/unit_stockpile_limit.lua

```diff
@@ -28,14 +28,19 @@ if gadgetHandler:IsSyncedCode() then
 	local GiveOrderToUnit	= Spring.GiveOrderToUnit
 
 	local canStockpile = {}
+	Spring.Log(gadget:GetInfo().name, LOG.WARNING, "Check all stockpiles "..#UnitDefs)
 	for udid, ud in pairs(UnitDefs) do
 		if ud.canStockpile then
 			canStockpile[udid] = true
 		end
-		if ud.customParams.stockpileLimit then
-			isStockpilingUnit[udid] = tonumber(ud.customParams.stockpileLimit)
-		elseif ud.customParams.stockpilelimit then
-			isStockpilingUnit[udid] = tonumber(ud.customParams.stockpilelimit)
+		if ud.weapons then
+			for i = 1, #ud.weapons do
+				local weaponDef = WeaponDefs[ud.weapons[i].weaponDef]
+				if weaponDef.stockpile and weaponDef.customParams and weaponDef.customParams.stockpilelimit then
+					Spring.Log(gadget:GetInfo().name, LOG.INFO, "Stockpile "..ud.name.." - "..weaponDef.name.." limit "..weaponDef.customParams.stockpilelimit)
+					isStockpilingUnit[udid] = tonumber(weaponDef.customParams.stockpilelimit)
```

Comment:
Table name should represent what it contains. Shouldn't the table be indexed by weapon def ID, not unit def ID?
```suggestion
					weaponStockpileLimits[weaponDef.weaponDef] = tonumber(weaponDef.customParams.stockpilelimit)
```
I have no idea why the engine calls the weapon def ID `weaponDef` instead of `weaponDefID`....

Reply:
It is a curious implementation, from what I can tell it needs to know the unit in order to give it orders to build more stockpile weapons. Probably unitStockpileLimits? At the moment it is very limited to one stockpile weapon per unit.